### PR TITLE
[4.0] Remove the deprecated code in the bootstrap files

### PR DESCRIPTION
**Pull request by laoneo**
Remove the deprecated code in the bootstrap files.

### CODE REVIEW

File: libraries/cms.php [hunk 1]
```diff
@@ -66,17 +66,3 @@
...
-JLoader::register('JInstallerFile',  JPATH_PLATFORM . '/cms/installer/adapter/file.php');
-JLoader::register('JInstallerLanguage',  JPATH_PLATFORM . '/cms/installer/adapter/language.php');
-JLoader::register('JInstallerLibrary',  JPATH_PLATFORM . '/cms/installer/adapter/library.php');
-JLoader::register('JInstallerModule',  JPATH_PLATFORM . '/cms/installer/adapter/module.php');
-JLoader::register('JInstallerPackage',  JPATH_PLATFORM . '/cms/installer/adapter/package.php');
-JLoader::register('JInstallerPlugin',  JPATH_PLATFORM . '/cms/installer/adapter/plugin.php');
-JLoader::register('JInstallerTemplate',  JPATH_PLATFORM . '/cms/installer/adapter/template.php');
-JLoader::register('JExtension',  JPATH_PLATFORM . '/cms/installer/extension.php');
-JLoader::registerAlias('JAdministrator',  'JApplicationAdministrator');
-JLoader::registerAlias('JSite',  'JApplicationSite');
```
Comment (wilsonge) [inline libraries/cms.php:82]: Some of these can cause potential issues. For example https://github.com/joomla/joomla-cms/issues/18761

Comment (laoneo) [reply libraries/cms.php:82]: I can test tomorrow if the search plugins do still work. but honestly for me there is no reason why this is needed at all in J3 or J4.

Comment (mbabker) [reply libraries/cms.php:82]: `JSite` and `JAdministrator` were renamed at 3.2, that's why it's needed in 3.x.

Comment (laoneo) [reply libraries/cms.php:82]: I was refering to the search plugins, if they are still using JSite. I'm aware that we can't remove it in 3.

**Comment by laoneo** [comment]
Removed the JSite hacks in the search component. I tried the searches in the front and gathered statistics for the back end. Never got an issue because of a missing JSite class.

**Comment by bonzani** [comment]
I have tested this item :white_check_mark: successfully on ba1097ee9be9b9e2f2628062809f7948f256933a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22353">issues.joomla.org/tracker/joomla-cms/22353</a>.</sub>

**Comment by wilsonge** [comment]
OK I've been meaning to test some parts of the search plugin but just haven't found the time so can we fix the conflicts here and i'll just merge it and we can find anything in testing in nightly's if there is anything

**Comment by laoneo** [comment]
Fixed

**Comment by wilsonge** [comment]
You're going to have to go back through this one I'm afraid. There's some places in the privacy component for example where we are using the old modal code for example

**Comment by wilsonge** [comment]
OK Spent some time looking into the search thing. I'd imagine that code is hacking around something like https://github.com/joomla/joomla-cms/blob/staging/plugins/search/content/content.php#L401 because if we get an admin menu instance here we're never going to be able to find an item id. It's just going to be about getting an SEF URL I guess. Also possible if we get https://github.com/joomla/joomla-cms/blob/staging/plugins/search/content/content.php#L53 out the container then we can force this to always be a site instance which is more sane anyhow.

**Comment by wilsonge** [comment]
Still unconvinced RE: the above. But we need to continue cleanup for beta. So we can always bring some form of that code back if essential

**Pull request merged**